tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BCompileUtil.java

```diff
@@ -149,41 +149,16 @@ private static void runInit(BLangPackage bLangPackage, JBallerinaInMemoryClassLo
                                                                bLangPackage.packageID.name.value,
                                                                TestConstant.MODULE_INIT_CLASS_NAME);
         Class<?> initClazz = classLoader.loadClass(initClassName);
+        final Scheduler scheduler = new Scheduler(4, false);
         if (temp) {
-            final Scheduler scheduler = new Scheduler(4, false);
```

Comment:
Shall we refactor code as below since we do not need runOnShedule inside if condition as per logic.
            runOnSchedule(initClazz, bLangPackage.initFunction.name, scheduler);
            runOnSchedule(initClazz, bLangPackage.startFunction.name, scheduler);
  if (temp) {
            scheduler.immortal = true;
            new Thread(scheduler::start).start();
        }